### PR TITLE
Make iBob documentation up to date.

### PR DESCRIPTION
**Pull request by sherfert**

### CODE REVIEW

**Review by tobias-johansson — APPROVED** [review]
:+1:

File: cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc [hunk 1]
```diff
@@ -179,120 +170,79 @@ Now consider the following refinement to the query:
...
-
-[source, cypher]
-----
-PROFILE
-MATCH (p:Person)-[:ACTED_IN]->(m:Movie) USING INDEX p:Person(name) WHERE exists(p.name)
-RETURN p.name, count(m) ORDER BY p.name
-----
-
-//profile
+In Neo4j 3.5 and later Cypher will recognise that the index already returns data in the correct order, and skip the `Sort` operation.
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc]:
```suggestion
In Neo4j 3.5 and later, Cypher will recognize that the index already returns data in the correct order, and skip the `Sort` operation.
```

File: cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc [hunk 2]
```diff
@@ -179,120 +170,79 @@ Now consider the following refinement to the query:
...
 
-[source, cypher]
-----
-PROFILE
-MATCH (p:Person)-[:ACTED_IN]->(m:Movie) WHERE p.name STARTS WITH ''
-RETURN min(p.name) AS name
-----
+Aggregations are usually using the `EagerAggregation` operation.
+This would mean scanning all nodes in the index to find the name that is first in alphabetic order.
+Instead, the query is planned with `Projection` followed by `Limit` followed by `Optional`.
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc]:
```suggestion
Instead, the query is planned with `Projection`, followed by `Limit`, followed by `Optional`.
```

File: cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc [hunk 3]
```diff
@@ -179,120 +170,79 @@ Now consider the following refinement to the query:
...
+* Several predicates combined using `OR`
+* Equality or range predicates querying for points (e.g. `WHERE n.place > point({ x: 1, y: 2 })`)
+* Spatial distance predicates (e.g. `WHERE distance(n.place, point({ x: 1, y: 2 })) < 2`)
+
 
 [NOTE]
 ====
-As of Neo4j 4.1, predicates with parameters, such as `WHERE n.prop > $param`, can trigger _index-backed order by_.
-The only exception are queries with any parameters of type `Float` or `Point`.
+If there is an existence constraint on the property, no predicate is required to trigger the optimization. For example, `CREATE CONSTRAINT constraint_name ON (p:Person) ASSERT exists(p.name)`
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc]:
```suggestion
If there is an existence constraint on the property, no predicate is required to trigger the optimization. 
For example, `CREATE CONSTRAINT constraint_name ON (p:Person) ASSERT exists(p.name)`
```

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @sherfert & @tobias-johansson 

I've pushed a few minor editorial changes.

File: cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc (hunk 1 above)
Comment (joangava) [reply cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc]: Morning il new user please i need yours help to continue

**Pull request merged**